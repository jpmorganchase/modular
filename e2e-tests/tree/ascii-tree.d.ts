declare module 'ascii-tree' {
  export function generate(input: string): string;
}
