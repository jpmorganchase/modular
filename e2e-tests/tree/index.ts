import fs from 'fs';
import path from 'path';
import asciiTree from 'ascii-tree';
import hash from '@emotion/hash';

function times(str: string, length: number) {
  return Array.from({ length }, () => str).join('');
}

const defaultIgnores = ['node_modules', '.git', '.DS_Store'];
const defaultHashIgnores = [
  // adding lockfiles here because it can be different
  // on different runs; since we install the latest versions
  // of some packages when making a repository
  'yarn.lock',
  'package-lock.json',
  // adding package.json files since versions within these get bumped
  'package.json',
];

function tree(
  _path: string,
  level = 1,
  options = {
    ignores: defaultIgnores,
    hashIgnores: defaultHashIgnores,
  },
): string {
  const stat = fs.statSync(_path);
  if (stat.isDirectory()) {
    const children = fs.readdirSync(_path);
    const dirArr = _path.split('/');
    const dir = dirArr[dirArr.length - 1];
    // todo - should these be sorted?
    // todo - handle symlinks, etc
    return `${times('#', level)}${dir}\n${children
      .filter((child: string) => !options.ignores.includes(child))
      .map((child: string) => tree(path.join(_path, child), level + 1, options))
      .join('\n')}`;
  } else {
    return [
      `${times('#', level)}${path.basename(_path)}`,
      options.hashIgnores.includes(path.basename(_path))
        ? undefined
        : `#${hash(fs.readFileSync(_path, 'utf8'))}`,
    ]
      .filter(Boolean)
      .join(' ');
  }
}

export default function generate(_path: string): string {
  return asciiTree.generate(tree(_path));
}
