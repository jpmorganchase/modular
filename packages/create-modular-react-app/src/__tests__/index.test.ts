/* eslint-disable jest/no-disabled-tests */
import createModularApp from '../';
import fs from 'fs-extra';
import path from 'path';
import tree from 'tree-view-for-tests';
import tmp from 'tmp';

// We want to omit any information that makes our snapshots
// fragile and therefore censor the author and package versions
// using `?`.
async function readCensoredPackageJson(
  packageJsonPath: string,
): Promise<unknown> {
  /* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return */
  function censorDiffering(packageJson: any): any {
    packageJson.author = '?';

    for (const dependencyTypeProperty of [
      'dependencies',
      'devDependencies',
      'peerDependencies',
      'optionalDependencies',
    ]) {
      if (packageJson[dependencyTypeProperty]) {
        // This replaces the version numbers of packages with `?`.
        packageJson[dependencyTypeProperty] = Object.fromEntries(
          Object.entries(packageJson[dependencyTypeProperty]).map(
            ([packageName]) => [packageName, '?'],
          ),
        );
      }
    }

    return packageJson;
  }

  return censorDiffering(await fs.readJson(packageJsonPath));
  /* eslint-enable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return */
}

describe.skip('create-modular-react-app', () => {
  const repoName = 'test-repo';
  let destination: string;
  let tmpDirectory: tmp.DirResult;
  beforeEach(() => {
    tmpDirectory = tmp.dirSync({ unsafeCleanup: true });
    destination = path.join(tmpDirectory.name, repoName);
  });

  afterEach(() => {
    //remove project
    tmpDirectory.removeCallback();
    // @ts-ignore
    destination = undefined;
  });
  it('should create a project with defaults', async () => {
    await createModularApp({ name: destination });
    expect(tree(destination)).toMatchInlineSnapshot(`
      "test-repo
      ├─ .editorconfig #1p4gvuw
      ├─ .eslintignore #1ot2bpo
      ├─ .gitignore #175wbq
      ├─ .prettierignore #10uqwgj
      ├─ .vscode
      │  ├─ extensions.json #1i4584r
      │  ├─ launch.json #1kk1omt
      │  └─ settings.json #xes41c
      ├─ .yarnrc #1orkcoz
      ├─ README.md #1nksyzj
      ├─ modular
      │  ├─ setupEnvironment.ts #m0s4vb
      │  └─ setupTests.ts #bnjknz
      ├─ package.json
      ├─ packages
      │  ├─ README.md #14bthrh
      │  └─ app
      │     ├─ package.json
      │     ├─ public
      │     │  ├─ favicon.ico #6pu3rg
      │     │  ├─ index.html #1m6toxd
      │     │  ├─ logo192.png #1nez7vk
      │     │  ├─ logo512.png #1hwqvcc
      │     │  ├─ manifest.json #19gah8o
      │     │  └─ robots.txt #1sjb8b3
      │     ├─ src
      │     │  ├─ App.css #1o0zosm
      │     │  ├─ App.tsx #c80ven
      │     │  ├─ __tests__
      │     │  │  └─ App.test.tsx #16urcos
      │     │  ├─ index.css #o7sk21
      │     │  ├─ index.tsx #zdn6mw
      │     │  ├─ logo.svg #1okqmlj
      │     │  └─ react-app-env.d.ts #t4ygcy
      │     └─ tsconfig.json #6rw46b
      ├─ tsconfig.json #1h72lkd
      └─ yarn.lock"
    `);
    expect(
      await readCensoredPackageJson(path.join(destination, 'package.json')),
    ).toMatchInlineSnapshot(`
      Object {
        "author": "?",
        "browserslist": Object {
          "development": Array [
            "last 1 chrome version",
            "last 1 firefox version",
            "last 1 safari version",
          ],
          "production": Array [
            ">0.2%",
            "not dead",
            "not op_mini all",
          ],
        },
        "dependencies": Object {
          "@testing-library/dom": "?",
          "@testing-library/jest-dom": "?",
          "@testing-library/react": "?",
          "@testing-library/user-event": "?",
          "@types/jest": "?",
          "@types/node": "?",
          "@types/react": "?",
          "@types/react-dom": "?",
          "eslint-config-modular-app": "?",
          "modular-scripts": "?",
          "prettier": "?",
          "react": "?",
          "react-dom": "?",
          "typescript": "?",
        },
        "eslintConfig": Object {
          "extends": "modular-app",
        },
        "license": "MIT",
        "main": "index.js",
        "modular": Object {
          "type": "root",
        },
        "name": "test-repo",
        "prettier": Object {
          "printWidth": 80,
          "proseWrap": "always",
          "singleQuote": true,
          "trailingComma": "all",
        },
        "private": true,
        "scripts": Object {
          "build": "modular build",
          "lint": "eslint . --ext .js,.ts,.tsx",
          "prettier": "prettier --write .",
          "start": "modular start",
          "test": "modular test",
        },
        "version": "1.0.0",
        "workspaces": Array [
          "packages/**",
        ],
      }
    `);
    expect(
      await readCensoredPackageJson(
        path.join(destination, 'packages', 'app', 'package.json'),
      ),
    ).toMatchInlineSnapshot(`
      Object {
        "author": "?",
        "dependencies": Object {},
        "modular": Object {
          "type": "app",
        },
        "name": "app",
        "private": true,
        "version": "0.1.0",
      }
    `);
  });
  it('should create a project with prefer offline', async () => {
    await createModularApp({ name: destination, preferOffline: true });
    expect(tree(destination)).toMatchInlineSnapshot(`
      "test-repo
      ├─ .editorconfig #1p4gvuw
      ├─ .eslintignore #1ot2bpo
      ├─ .gitignore #175wbq
      ├─ .prettierignore #10uqwgj
      ├─ .vscode
      │  ├─ extensions.json #1i4584r
      │  ├─ launch.json #1kk1omt
      │  └─ settings.json #xes41c
      ├─ .yarnrc #1orkcoz
      ├─ README.md #1nksyzj
      ├─ modular
      │  ├─ setupEnvironment.ts #m0s4vb
      │  └─ setupTests.ts #bnjknz
      ├─ package.json
      ├─ packages
      │  ├─ README.md #14bthrh
      │  └─ app
      │     ├─ package.json
      │     ├─ public
      │     │  ├─ favicon.ico #6pu3rg
      │     │  ├─ index.html #1m6toxd
      │     │  ├─ logo192.png #1nez7vk
      │     │  ├─ logo512.png #1hwqvcc
      │     │  ├─ manifest.json #19gah8o
      │     │  └─ robots.txt #1sjb8b3
      │     ├─ src
      │     │  ├─ App.css #1o0zosm
      │     │  ├─ App.tsx #c80ven
      │     │  ├─ __tests__
      │     │  │  └─ App.test.tsx #16urcos
      │     │  ├─ index.css #o7sk21
      │     │  ├─ index.tsx #zdn6mw
      │     │  ├─ logo.svg #1okqmlj
      │     │  └─ react-app-env.d.ts #t4ygcy
      │     └─ tsconfig.json #6rw46b
      ├─ tsconfig.json #1h72lkd
      └─ yarn.lock"
    `);
    expect(
      await readCensoredPackageJson(path.join(destination, 'package.json')),
    ).toMatchInlineSnapshot(`
      Object {
        "author": "?",
        "browserslist": Object {
          "development": Array [
            "last 1 chrome version",
            "last 1 firefox version",
            "last 1 safari version",
          ],
          "production": Array [
            ">0.2%",
            "not dead",
            "not op_mini all",
          ],
        },
        "dependencies": Object {
          "@testing-library/dom": "?",
          "@testing-library/jest-dom": "?",
          "@testing-library/react": "?",
          "@testing-library/user-event": "?",
          "@types/jest": "?",
          "@types/node": "?",
          "@types/react": "?",
          "@types/react-dom": "?",
          "eslint-config-modular-app": "?",
          "modular-scripts": "?",
          "prettier": "?",
          "react": "?",
          "react-dom": "?",
          "typescript": "?",
        },
        "eslintConfig": Object {
          "extends": "modular-app",
        },
        "license": "MIT",
        "main": "index.js",
        "modular": Object {
          "type": "root",
        },
        "name": "test-repo",
        "prettier": Object {
          "printWidth": 80,
          "proseWrap": "always",
          "singleQuote": true,
          "trailingComma": "all",
        },
        "private": true,
        "scripts": Object {
          "build": "modular build",
          "lint": "eslint . --ext .js,.ts,.tsx",
          "prettier": "prettier --write .",
          "start": "modular start",
          "test": "modular test",
        },
        "version": "1.0.0",
        "workspaces": Array [
          "packages/**",
        ],
      }
    `);
    expect(
      await readCensoredPackageJson(
        path.join(destination, 'packages', 'app', 'package.json'),
      ),
    ).toMatchInlineSnapshot(`
      Object {
        "author": "?",
        "dependencies": Object {},
        "modular": Object {
          "type": "app",
        },
        "name": "app",
        "private": true,
        "version": "0.1.0",
      }
    `);
  });
  it('should create a project without git metadata', async () => {
    await createModularApp({
      name: destination,
      repo: false,
    });
    expect(fs.existsSync(path.join(destination, '.git'))).toEqual(false);
  });
  it('should create a project prefer offline without git metadata', async () => {
    await createModularApp({
      name: destination,
      repo: false,
      preferOffline: true,
    });
    expect(fs.existsSync(path.join(destination, '.git'))).toEqual(false);
  });
});
