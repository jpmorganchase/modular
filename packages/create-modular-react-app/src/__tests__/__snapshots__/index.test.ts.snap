// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-modular-react-app WHEN it sets up a project with prefer Offline SHOULD setup an app package.json correctly: 
        Object {
          "author": "?",
          "dependencies": Object {},
          "modular": Object {
            "type": "app",
          },
          "name": "app",
          "private": true,
          "version": "0.1.0",
        }
       1`] = `
{
  "author": "?",
  "modular": {
    "type": "app",
  },
  "name": "app",
  "private": true,
  "version": "1.0.0",
}
`;

exports[`create-modular-react-app WHEN it sets up a project with prefer Offline SHOULD setup an package.json correctly: 
        Object {
          "author": "?",
          "browserslist": Object {
            "development": Array [
              "last 1 chrome version",
              "last 1 firefox version",
              "last 1 safari version",
            ],
            "production": Array [
              ">0.2%",
              "not dead",
              "not op_mini all",
            ],
          },
          "dependencies": Object {
            "@testing-library/dom": "?",
            "@testing-library/jest-dom": "?",
            "@testing-library/react": "?",
            "@testing-library/user-event": "?",
            "@types/jest": "?",
            "@types/node": "?",
            "@types/react": "?",
            "@types/react-dom": "?",
            "eslint-config-modular-app": "?",
            "modular-scripts": "?",
            "prettier": "?",
            "react": "?",
            "react-dom": "?",
            "typescript": "?",
          },
          "eslintConfig": Object {
            "extends": "modular-app",
          },
          "license": "MIT",
          "main": "index.js",
          "modular": Object {
            "type": "root",
          },
          "name": "test-repo",
          "prettier": Object {
            "printWidth": 80,
            "proseWrap": "always",
            "singleQuote": true,
            "trailingComma": "all",
          },
          "private": true,
          "scripts": Object {
            "build": "modular build",
            "lint": "eslint . --ext .js,.ts,.tsx",
            "prettier": "prettier --write .",
            "start": "modular start",
            "test": "modular test",
          },
          "version": "1.0.0",
          "workspaces": Array [
            "packages/**",
          ],
        }
       1`] = `
{
  "author": "?",
  "browserslist": {
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version",
    ],
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all",
    ],
  },
  "dependencies": {
    "@testing-library/dom": "?",
    "@testing-library/jest-dom": "?",
    "@testing-library/react": "?",
    "@testing-library/user-event": "?",
    "@types/jest": "?",
    "@types/node": "?",
    "@types/react": "?",
    "@types/react-dom": "?",
    "eslint-config-modular-app": "?",
    "modular-scripts": "?",
    "modular-template-app": "?",
    "prettier": "?",
    "react": "?",
    "react-dom": "?",
    "typescript": "?",
  },
  "eslintConfig": {
    "extends": "modular-app",
  },
  "license": "MIT",
  "main": "index.js",
  "modular": {
    "type": "root",
  },
  "name": "test-repo",
  "prettier": {
    "printWidth": 80,
    "proseWrap": "always",
    "singleQuote": true,
    "trailingComma": "all",
  },
  "private": true,
  "scripts": {
    "build": "modular build",
    "lint": "eslint . --ext .js,.ts,.tsx",
    "prettier": "prettier --write .",
    "start": "modular start",
    "test": "modular test",
  },
  "version": "1.0.0",
  "workspaces": [
    "packages/**",
  ],
}
`;

exports[`create-modular-react-app WHEN it sets up a project with prefer Offline should create a project with prefer offline: 
        "test-repo
        ├─ .editorconfig #1p4gvuw
        ├─ .eslintignore #1ot2bpo
        ├─ .gitignore #175wbq
        ├─ .prettierignore #10uqwgj
        ├─ .vscode
        │  ├─ extensions.json #1i4584r
        │  ├─ launch.json #1kk1omt
        │  └─ settings.json #xes41c
        ├─ .yarnrc #1orkcoz
        ├─ README.md #1nksyzj
        ├─ modular
        │  ├─ setupEnvironment.ts #m0s4vb
        │  └─ setupTests.ts #bnjknz
        ├─ package.json
        ├─ packages
        │  ├─ README.md #14bthrh
        │  └─ app
        │     ├─ package.json
        │     ├─ public
        │     │  ├─ favicon.ico #6pu3rg
        │     │  ├─ index.html #1m6toxd
        │     │  ├─ logo192.png #1nez7vk
        │     │  ├─ logo512.png #1hwqvcc
        │     │  ├─ manifest.json #19gah8o
        │     │  └─ robots.txt #1sjb8b3
        │     ├─ src
        │     │  ├─ App.css #1o0zosm
        │     │  ├─ App.tsx #c80ven
        │     │  ├─ __tests__
        │     │  │  └─ App.test.tsx #16urcos
        │     │  ├─ index.css #o7sk21
        │     │  ├─ index.tsx #zdn6mw
        │     │  ├─ logo.svg #1okqmlj
        │     │  └─ react-app-env.d.ts #t4ygcy
        │     └─ tsconfig.json #6rw46b
        ├─ tsconfig.json #1h72lkd
        └─ yarn.lock"
       1`] = `
"test-repo
├─ .editorconfig #1p4gvuw
├─ .eslintignore #1ot2bpo
├─ .gitignore #175wbq
├─ .prettierignore #10uqwgj
├─ .vscode
│  ├─ extensions.json #1i4584r
│  ├─ launch.json #1kk1omt
│  └─ settings.json #xes41c
├─ .yarnrc #1orkcoz
├─ README.md #5vircs
├─ modular
│  ├─ setupEnvironment.ts #m0s4vb
│  └─ setupTests.ts #bnjknz
├─ package.json
├─ packages
│  ├─ README.md #11apaz5
│  └─ app
│     ├─ package.json
│     ├─ public
│     │  ├─ favicon.ico #6pu3rg
│     │  ├─ index.html #1m6toxd
│     │  ├─ logo192.png #1nez7vk
│     │  ├─ logo512.png #1hwqvcc
│     │  ├─ manifest.json #19gah8o
│     │  └─ robots.txt #1sjb8b3
│     ├─ src
│     │  ├─ App.css #1o0zosm
│     │  ├─ App.tsx #c80ven
│     │  ├─ __tests__
│     │  │  └─ App.test.tsx #16urcos
│     │  ├─ index.css #o7sk21
│     │  ├─ index.tsx #zdn6mw
│     │  ├─ logo.svg #1okqmlj
│     │  └─ react-app-env.d.ts #t4ygcy
│     └─ tsconfig.json #6rw46b
├─ tsconfig.json #1h72lkd
└─ yarn.lock"
`;

exports[`create-modular-react-app WHEN setting a project with defaults Sets up an app package.json correctly: 
          Object {
            "author": "?",
            "dependencies": Object {},
            "modular": Object {
              "type": "app",
            },
            "name": "app",
            "private": true,
            "version": "0.1.0",
          }
         1`] = `
{
  "author": "?",
  "modular": {
    "type": "app",
  },
  "name": "app",
  "private": true,
  "version": "1.0.0",
}
`;

exports[`create-modular-react-app WHEN setting a project with defaults Sets up the package.json correctly: 
          Object {
            "author": "?",
            "browserslist": Object {
              "development": Array [
                "last 1 chrome version",
                "last 1 firefox version",
                "last 1 safari version",
              ],
              "production": Array [
                ">0.2%",
                "not dead",
                "not op_mini all",
              ],
            },
            "dependencies": Object {
              "@testing-library/dom": "?",
              "@testing-library/jest-dom": "?",
              "@testing-library/react": "?",
              "@testing-library/user-event": "?",
              "@types/jest": "?",
              "@types/node": "?",
              "@types/react": "?",
              "@types/react-dom": "?",
              "eslint-config-modular-app": "?",
              "modular-scripts": "?",
              "prettier": "?",
              "react": "?",
              "react-dom": "?",
              "typescript": "?",
            },
            "eslintConfig": Object {
              "extends": "modular-app",
            },
            "license": "MIT",
            "main": "index.js",
            "modular": Object {
              "type": "root",
            },
            "name": "test-repo",
            "prettier": Object {
              "printWidth": 80,
              "proseWrap": "always",
              "singleQuote": true,
              "trailingComma": "all",
            },
            "private": true,
            "scripts": Object {
              "build": "modular build",
              "lint": "eslint . --ext .js,.ts,.tsx",
              "prettier": "prettier --write .",
              "start": "modular start",
              "test": "modular test",
            },
            "version": "1.0.0",
            "workspaces": Array [
              "packages/**",
            ],
          }
         1`] = `
{
  "author": "?",
  "browserslist": {
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version",
    ],
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all",
    ],
  },
  "dependencies": {
    "@testing-library/dom": "?",
    "@testing-library/jest-dom": "?",
    "@testing-library/react": "?",
    "@testing-library/user-event": "?",
    "@types/jest": "?",
    "@types/node": "?",
    "@types/react": "?",
    "@types/react-dom": "?",
    "eslint-config-modular-app": "?",
    "modular-scripts": "?",
    "modular-template-app": "?",
    "prettier": "?",
    "react": "?",
    "react-dom": "?",
    "typescript": "?",
  },
  "eslintConfig": {
    "extends": "modular-app",
  },
  "license": "MIT",
  "main": "index.js",
  "modular": {
    "type": "root",
  },
  "name": "test-repo",
  "prettier": {
    "printWidth": 80,
    "proseWrap": "always",
    "singleQuote": true,
    "trailingComma": "all",
  },
  "private": true,
  "scripts": {
    "build": "modular build",
    "lint": "eslint . --ext .js,.ts,.tsx",
    "prettier": "prettier --write .",
    "start": "modular start",
    "test": "modular test",
  },
  "version": "1.0.0",
  "workspaces": [
    "packages/**",
  ],
}
`;

exports[`create-modular-react-app WHEN setting a project with defaults should create a project with defaults: 
          "test-repo
          ├─ .editorconfig #1p4gvuw
          ├─ .eslintignore #1ot2bpo
          ├─ .gitignore #175wbq
          ├─ .prettierignore #10uqwgj
          ├─ .vscode
          │  ├─ extensions.json #1i4584r
          │  ├─ launch.json #1kk1omt
          │  └─ settings.json #xes41c
          ├─ .yarnrc #1orkcoz
          ├─ README.md #1nksyzj
          ├─ modular
          │  ├─ setupEnvironment.ts #m0s4vb
          │  └─ setupTests.ts #bnjknz
          ├─ package.json
          ├─ packages
          │  ├─ README.md #14bthrh
          │  └─ app
          │     ├─ package.json
          │     ├─ public
          │     │  ├─ favicon.ico #6pu3rg
          │     │  ├─ index.html #1m6toxd
          │     │  ├─ logo192.png #1nez7vk
          │     │  ├─ logo512.png #1hwqvcc
          │     │  ├─ manifest.json #19gah8o
          │     │  └─ robots.txt #1sjb8b3
          │     ├─ src
          │     │  ├─ App.css #1o0zosm
          │     │  ├─ App.tsx #c80ven
          │     │  ├─ __tests__
          │     │  │  └─ App.test.tsx #16urcos
          │     │  ├─ index.css #o7sk21
          │     │  ├─ index.tsx #zdn6mw
          │     │  ├─ logo.svg #1okqmlj
          │     │  └─ react-app-env.d.ts #t4ygcy
          │     └─ tsconfig.json #6rw46b
          ├─ tsconfig.json #1h72lkd
          └─ yarn.lock"
         1`] = `
"test-repo
├─ .editorconfig #1p4gvuw
├─ .eslintignore #1ot2bpo
├─ .gitignore #175wbq
├─ .prettierignore #10uqwgj
├─ .vscode
│  ├─ extensions.json #1i4584r
│  ├─ launch.json #1kk1omt
│  └─ settings.json #xes41c
├─ .yarnrc #1orkcoz
├─ README.md #5vircs
├─ modular
│  ├─ setupEnvironment.ts #m0s4vb
│  └─ setupTests.ts #bnjknz
├─ package.json
├─ packages
│  ├─ README.md #11apaz5
│  └─ app
│     ├─ package.json
│     ├─ public
│     │  ├─ favicon.ico #6pu3rg
│     │  ├─ index.html #1m6toxd
│     │  ├─ logo192.png #1nez7vk
│     │  ├─ logo512.png #1hwqvcc
│     │  ├─ manifest.json #19gah8o
│     │  └─ robots.txt #1sjb8b3
│     ├─ src
│     │  ├─ App.css #1o0zosm
│     │  ├─ App.tsx #c80ven
│     │  ├─ __tests__
│     │  │  └─ App.test.tsx #16urcos
│     │  ├─ index.css #o7sk21
│     │  ├─ index.tsx #zdn6mw
│     │  ├─ logo.svg #1okqmlj
│     │  └─ react-app-env.d.ts #t4ygcy
│     └─ tsconfig.json #6rw46b
├─ tsconfig.json #1h72lkd
└─ yarn.lock"
`;
